apps/backend/src/lambdas/parseTDI.ts

```diff
@@ -0,0 +1,99 @@
+import { NestFactory } from '@nestjs/core';
+import { Context } from 'aws-lambda';
+import { AppModule } from '../app.module';
+import { AppLogger } from '../common/logger.service';
+import { S3ManagerService } from '../s3-manager/s3-manager.service';
+import { TDI17Record } from '../tdi-parser/tdi17/TDI17Record';
+import { TDI17Header } from '../tdi-parser/tdi17/TDI17Header';
+import { TDI17Details } from '../tdi-parser/tdi17/TDI17Details';
+import { TDI17Trailer } from '../tdi-parser/tdi17/TDI17Trailer';
+import { TDI34Record } from '../tdi-parser/tdi34/TDI34Record';
+import { TDI34Header } from '../tdi-parser/tdi34/TDI34Header';
+import { TDI34Details } from '../tdi-parser/tdi34/TDI34Details';
+import { TDI34Trailer } from '../tdi-parser/tdi34/TDI34Trailer';
+import { TDI34Enum } from './const/tdi34.const';
+import { TDI17Enum } from './const/tdi17.const';
+
+export const handler = async (
+  event?: { filePath: string; type: string },
+  context?: Context,
+) => {
+  const app = await NestFactory.createApplicationContext(AppModule);
+  const appLogger = app.get(AppLogger);
+  const s3manager = app.get(S3ManagerService);
+
+  appLogger.log({ event });
+  appLogger.log({ context });
+
+  const filetype = event?.type ?? process.env.TDI_TYPE;
+  const TDIEnum = filetype === 'TDI17' ? TDI17Enum : TDI34Enum;
+
+  try {
+    appLogger.log('...start TDI Parsing');
+    const contents = await s3manager.getContents(
+      TDIEnum.BUCKET,
+      event?.filePath ?? TDIEnum.LOCAL_PATH,
+    );
+
+    const record = contents?.Body?.toString() || '';
+    const output = parseTDI(record, TDIEnum);
+
+    appLogger.log(output);
+    const outputFile = Buffer.from(JSON.stringify(output));
+
+    appLogger.log(outputFile);
+    await s3manager.putObject(
+      TDIEnum.BUCKET,
+      TDIEnum.OUTPUT_PATH + Date.now() + TDIEnum.JSON_FILENAME,
+      outputFile,
+      TDIEnum.JSON_TYPE,
+    );
+  } catch (e) {
+    appLogger.error(e);
+  }
+};
+
+const parseTDI = (fileContents: string, TDIEnum: unknown) => {
+  const TDI_17 = TDIEnum === TDI17Enum;
+
+  const lines = fileContents.split('\n').filter((l: string) => l);
+
+  const header = TDI_17 ? new TDI17Header({}) : new TDI34Header({});
+  const trailer = TDI_17 ? new TDI17Trailer({}) : new TDI34Trailer({});
```

Comment:
I don't think we want to store the header and trailer for TDI files. 

The transactions line items is all we want.